store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/utils/AggregateGafferRowsFunction.java

```diff
@@ -0,0 +1,135 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.gaffer.parquetstore.utils;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import org.apache.spark.api.java.function.Function2;
+import org.apache.spark.sql.catalyst.expressions.GenericRowWithSchema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.gchq.gaffer.data.element.Properties;
+import uk.gov.gchq.gaffer.data.element.function.ElementAggregator;
+import uk.gov.gchq.gaffer.exception.SerialisationException;
+import uk.gov.gchq.gaffer.operation.OperationException;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.BinaryOperator;
+
+/**
+ *
+ */
+public class AggregateGafferRowsFunction implements Function2<GenericRowWithSchema, GenericRowWithSchema, GenericRowWithSchema>, Serializable {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AggregateGafferRowsFunction.class);
+    private static final long serialVersionUID = -8353767193380574516L;
+    private final Boolean isEntity;
+    private final HashMap<String, String> propertyToAggregatorMap;
+    private final Set<String> groupByColumns;
+    private final GafferGroupObjectConverter objectConverter;
+    private final HashMap<String, String[]> columnToPaths;
+    private final String[] gafferProperties;
+
+    @SuppressFBWarnings("EI_EXPOSE_REP2")
+    public AggregateGafferRowsFunction(final String[] gafferProperties, final boolean isEntity,
+                                       final Set<String> groupByColumns,
+                                       final HashMap<String, String[]> columnToPaths,
+                                       final HashMap<String, String> propertyToAggregatorMap,
+                                       final GafferGroupObjectConverter gafferGroupObjectConverter)
+            throws SerialisationException {
+        LOGGER.debug("Generating a new AggregateGafferRowsFunction");
+        this.gafferProperties = gafferProperties;
+        LOGGER.debug("gafferProperties:" + Arrays.toString(this.gafferProperties));
+        this.columnToPaths = columnToPaths;
+        this.objectConverter = gafferGroupObjectConverter;
+        this.isEntity = isEntity;
+        this.groupByColumns = groupByColumns;
+        LOGGER.debug("GroupByColumns:" + this.groupByColumns.toString());
+        this.propertyToAggregatorMap = propertyToAggregatorMap;
+        LOGGER.debug("PropertyToAggregatorMap:" + this.propertyToAggregatorMap.toString());
```

Comment:
also change this to:
LOGGER.debug("PropertyToAggregatorMap: {}", this.propertyToAggregatorMap);

Please could you go through your code and change for this sort of issue.